pkg/cache/cache.go

```diff
@@ -188,14 +191,69 @@ func BuilderWithOptions(options Options) NewCacheFunc {
 		if options.Namespace == "" {
 			options.Namespace = opts.Namespace
 		}
-		if opts.Resync == nil {
-			opts.Resync = options.Resync
+		for obj, selector := range opts.SelectorsByObject {
```

Comment:
I wasn't sure if it was intentional that we didn't inherit some of those field from the original options, but yeah, it caught me off guard when my tests were failing.

Sounds like at least you and I consider it a bug that we don't handle defaulting for everything in `cache.Options`, so I can make a separate PR for just this. 

As an aside, I'm not convinced I got the defaulting right for `TransformByObject` and `DefaultTransform`. We might want to try to combine those as well.

Reply:
Looks like this is a bug, but I'm not sure if the two options might have conflicts like one selects `K=V1` and the other is `K=V2`?

On the other hand, these optional fields like `SelectorsByObject` or `TransformByObject` can only be set via `BuilderWithOptions` if the cache is newed by manager. And if users use `cache.New` directly, they don't have to use `BuilderWithOptions`.

So I'm thinking how about make the two options different priorities, so that if an option like `SelectorsByObject` is set in both options, only one of them will be used?

Reply:
I broke out these changes into a separate PR here: https://github.com/kubernetes-sigs/controller-runtime/pull/1980